src/message-store-sql.ts

```diff
@@ -187,20 +234,25 @@ export class MessageStoreSql implements MessageStore {
 
     options?.signal?.throwIfAborted();
 
+    // extract sort property and direction from the supplied messageSort
+    const { property: sortProperty, direction: sortDirection } = this.extractSortProperties(messageSort);
+
     let query = this.#db
       .selectFrom('messageStore')
-      .selectAll()
+      .leftJoin('messageStoreRecordsTags', 'messageStoreRecordsTags.messageStoreId', 'messageStore.id')
+      .leftJoin('messageStoreRecordsTagValues', 'messageStoreRecordsTagValues.tagId', 'messageStoreRecordsTags.id')
```

Comment:
I went back and forth with this a bit when thinking about it, definitely wouldn't mind a short discussion to land on something.

I landed on the tags + values tables because it felt more normalized and I wasn't sure that there would be a huge performance hit.

But you may be right given that the tag values table will grow pretty large.

However, since we don't do any queries where we are looking at the `value` of the tag without knowing the actual tag's field I don't really know that a flat tags + values table will have a real effect.

Reply:
Let's chat!

Reply:
Done! Quite a bit less code, so that's always better!